modules/react-components/src/components/header/header.tsx

```diff
@@ -385,7 +385,22 @@ export const Header: FunctionComponent<HeaderPropsInterface> = (
             return null;
         }
 
-        return userDropdownLinks.map((category: HeaderLinkCategoryInterface, categoryIndex: number) => {
+        const adjustedUserDropdownLinks : HeaderLinkCategoryInterface[] = userDropdownLinks
+            .reduce((previousValue, currentValue) => {
+                const { category, categoryLabel, links } : Partial<HeaderLinkCategoryInterface> = currentValue;
+                const findObj : Partial<HeaderLinkCategoryInterface> = previousValue
```

Comment:
Let's take a shallow copy of the object, because asigning will only create a reference.
```suggestion
                const findObj : Partial<HeaderLinkCategoryInterface> = { ...previousValue }
```

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/2893